src/mqt/problemsolver/equivalence_checker/equivalence_checker.py

```diff
@@ -0,0 +1,235 @@
+    """Creates a synthetic miter string with multiple variables in form of letters.
+
+    These represent the input bits of to be verified circuits and can be either 0 or 1.
+    The function also returns the corresponding counter examples to the miter string, that is, the bitstrings that satisfy the miter condition.
+    A miter can be of the form "a & b & c & d" where a, b, c, d are input bits. The counter examples are the bitstrings that satisfy the miter condition, e.g. "0000" for the miter "a & b & c & d".
+
+    Parameters
+    ----------
+    num_bits : int
+        Number of input bits
+    num_counter_examples : int
+        Number of counter examples
+
+    Returns:
+    -------
+    res_string : str
+        Resulting condition string
+    counter_examples : list[str]
+        The corresponding bitstrings to res_string (e.g. counter_examples is ['1111'] for res_string 'a & b & c & d')
+    """
+    if num_bits < 0 or num_counter_examples < 0:
+        msg = "The number of bits or counter examples cannot be used."
+        raise ValueError(msg)
+
+    alphabet = list(string.ascii_lowercase)
+    counter_examples: list[str] = []
+    if (
+        num_counter_examples == 0
+    ):  # Since the qiskit PhaseOracle does not support empty conditions, we need to add a condition that is always false
+        res, _ = create_condition_string(num_bits, 1)  # returns e.g. "~a & ~b & ~c & ~d" for num_bits = 4
+        res += " & a"  # turns the miter into "~a & ~b & ~c & ~d & a" which is always false
+        return res, counter_examples
+    res_string: str = ""
+    counter_examples = []
+    for num in range(num_counter_examples):
+        bitstring = list(str(format(num, f"0{num_bits}b")))[
+            ::-1
+        ]  # e.g. ['0', '0', '0', '0'] for num = 0 and num_bits = 4
+        counter_examples.append(str(format(num, f"0{num_bits}b")))  # appends ['0000'] for num = 0 and num_bits = 4
+        for i, char in enumerate(bitstring):  # the following lines add a negated letter (e.g. "~a") for
+            if char == "0" and i == 0:  # each "0" and a letter (e.g. "a") for each "1" in the bitstring list.
+                bitstring[i] = (
+                    "~" + alphabet[i]
+                )  # If the first letter is added (so i > 0), the subsequent letters are added together
+            elif char == "1" and i == 0:  # with a logical AND operator (e.g. "~a & ~b & ~c & ~d").
+                bitstring[i] = alphabet[i]
+            elif char == "0":
+                bitstring[i] = " & " + "~" + alphabet[i]
+            elif char == "1":
+                bitstring[i] = " & " + alphabet[i]
+        combined_bitstring = "".join(bitstring)
+        if num < num_counter_examples - 1:
+            res_string += combined_bitstring + " | "
+        else:
+            res_string += combined_bitstring
+    return res_string, counter_examples
+
+
+miter, counter = create_condition_string(4, 2)
```

Comment:
Why is `create_condition_string` called here?

Reply:
Done